Don't allow PRTB namespace and the project ID of projectName to differ (#292)
rancher/webhook: pkg/resources/management.cattle.io/v3/projectroletemplatebinding/ProjectRoleTemplateBinding.md

Users cannot create ProjectRoleTemplateBindings that violate the following constraints:

> - The `ProjectName` field must be specified and have the format of `cluster-id:project-id`, and `project-id` must match
>   the namespace of the ProjectRoleTemplateBinding.

Reviewer: Nit: Consider fixing the grammar so that you only use one and, like:
"The `ProjectName` field must be specified, have the format of `cluster-id:project-id`, and `project-id` must match
  the namespace of the ProjectRoleTemplateBinding."

However, since there are now 3 requirements, I would go with something like what we did for RoleTemplate:
```suggestion
- The `ProjectName` field must be :
  - Provided as a non-empty value
  - Specified using the format of `cluster-id:project-id`
    - `project-id`, in particular, must match the namespace of the ProjectRoleTemplateBinding
```

Author: Adjusted.

Users cannot create ProjectRoleTemplateBindings that violate the following constraints:

> - The projectName field must have the format `cluster-id:project-id`, and `project-id` must match the namespace of the

Reviewer 2: This is talking about the same `ProjectName` as mentioned on line 17 right? Maybe state both requirements ("must be specified" and "have the format...") on the same line.

Author: Adjusted!

  the namespace of the ProjectRoleTemplateBinding.
- Either a user subject (through `UserName` or `UserPrincipalName`), or a group subject (through `GroupName`
  or `GroupPrincipalName`), or a service account subject (through `ServiceAccount`) must be specified. Exactly one
  subject type of the three must be provided.

> - `ProjectName` must be specified.

Reviewer 2: This is now duplicating the information on line 12

Users cannot create ProjectRoleTemplateBindings that violate the following constraints:

> - The namespace and project ID (the second part of the projectName field, as in `c-some-cluster:p-cgtq4`) must be

Reviewer: ```suggestion
- The projectName field must have the format `cluster-id:project-id`, and `project-id` must match the namespace of the ProjectRoleTemplateBinding.
```
